docs/GUIDE.md

```diff
@@ -0,0 +1,142 @@
+
+`true` enables additional check for possible missing codec registrations (more info about this problem in [README](../README.md)). This is in general checked by the [Codec Registration Checker Compiler Plugin](#codec-registration-checker-compiler-plugin-guide) during compilation, so the default is `false`. However, if you use only Circe Akka Serializer without compiler plugins - this check should be enabled. Be aware that enabling this check might have a slight negative impact on Serializer's performance.<br><br>
+
+- `org.virtuslab.ash.circe.compression.algorithm` = `gzip` / `off`
+
+`gzip` enables compression of payloads before serialization using the [Gzip](https://www.gnu.org/software/gzip) algorithm. Default is `off` (compression disabled). Choosing `gzip` altogether with setting proper `compress-larger-than` value (explained below) enables compression of payloads.<br><br>
+
+- `org.virtuslab.ash.circe.compression.compress-larger-than` = `64 KiB` _(example value)_
+
+If `org.virtuslab.ash.circe.compression.algorithm` = `gzip` and `compress-larger-than` **value** is larger than `0 KiB` &mdash; each payload greater than the **value** will be compressed before serialization. Default **value** is `32 KiB`. If `org.virtuslab.ash.circe.compression.algorithm` = `off` &mdash; payloads will not be compressed, regardless of the chosen `compress-larger-than` value.<br><br>
+**Note** &mdash; Circe Akka Serializer will perform **deserialization** properly for both compressed and not compressed payloads regardless of both `org.virtuslab.ash.circe.compression` configurations (Serializer will recognize whether the payload has been compressed programmatically). These settings matter only for choosing type of serialization.
+
+### Annotations Guide
+ASH compiler plugins are based on usages of two annotations: [@SerializabilityTrait](#SerializabilityTrait) and [@Serializer](#Serializer). Thus, before enabling compiler plugins, make sure that you are using these two annotations properly in the project / module where plugins will do their work. In order to have access to these annotations, add this part of ASH to library dependencies:
+```sbt
+import org.virtuslab.ash.AkkaSerializationHelperPlugin
+(...)
+libraryDependencies += AkkaSerializationHelperPlugin.annotation
+```
+#### SerializabilityTrait
+[@SerializabilityTrait](../annotation/src/main/scala/org/virtuslab/ash/annotation/SerializabilityTrait.scala) is an annotation that should be added to your top-level serializable type (trait mentioned in the `akka.actor.serialization-bindings` part of akka config) that is extended by Messages / Events / States. Moreover, if your top-level serializable type is extended by another **trait** &mdash; this trait should also be annotated with this annotation. Concrete classes extending "serializability traits" (i.e. classes that define Messages/Events/States) should not be marked with this annotation. See examples in [README.md](../README.md)
```

Comment:
```suggestion
[@SerializabilityTrait](../annotation/src/main/scala/org/virtuslab/ash/annotation/SerializabilityTrait.scala) is an annotation that should be added to your top-level serializable type (trait mentioned in the `akka.actor.serialization-bindings` part of Akka config) that is extended by Messages / Events / States. Moreover, if your top-level serializable type is extended by another **trait** &mdash; this trait should also be annotated with this annotation. Concrete classes extending "serializability traits" (i.e. classes that define Messages/Events/States) should not be marked with this annotation. See examples in [README.md](../README.md)
```

Reply:
> Moreover, if your top-level serializable type is extended by another **trait** &mdash; this trait should also be annotated with this annotation

Hmm what case do you refer to here?

Reply:
> See examples in [README.md](../README.md)

Pls provide link to the specific section

Reply:
> Hmm what case do you refer to here?

Example case (similar to what we have in Hydra now):
- the most top-level serializable trait (marker trait mentioned in the akka .conf file) is `trait TopSerializable`
- `TopSerializable` is extended by `trait SpecificSerializable`
- `SpecificSerializable` is extended by concrete classes (used as messages / events / states)
- `SpecificSerializable` is used in the `@Serializer` annotation
In such case both `TopSerializable` and `SpecificSerializable` should be marked with `@SerializabilityTrait`

But after writing this example I would say that I should add better explanation in this fragment of guide.